x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlResponseHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.authc.saml;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.opensaml.saml.saml2.core.Status;
+import org.opensaml.saml.saml2.core.StatusCode;
+import org.opensaml.saml.saml2.core.StatusDetail;
+import org.opensaml.saml.saml2.core.StatusMessage;
+import org.opensaml.saml.saml2.core.StatusResponseType;
+
+import java.time.Clock;
+import java.util.Collection;
+
+import static org.elasticsearch.xpack.security.authc.saml.SamlUtils.samlException;
+
+public class SamlResponseHandler extends SamlObjectHandler {
```

Comment:
This is mainly because we do have a real Request object from idP, i.e.`SamlInvalidateSessionRequest`. Its associated handler is `SamlLogoutRequestHandler`, which is a subclass of `SamlObjectHandler`. In another word, `SamlObjectHandler` is the base class for both Request and Response handlers. So I did not merge `SamlResponseHandler` into `SamlObjectHandler`.

Also I did a few refactoring which is likely made the changes hard to track, here is a summary:
* `SamlObjectHandler` was renamed from `SamlRequestHandler` (It seems to be a good idea to me since its the base for both Request and Response handlers. But I can revert it if it does not sound right to you).
* `SamlResponseHandler` is extracted from `SamlAuthenticator` since many code can be reused by the new `SamlLogoutResponseHandler`

The class hierarchy is as follows:
```
SamlObjectHandler
├── SamlLogoutRequestHandler
└── SamlResponseHandler
    ├── SamlAuthenticator
    └── SamlLogoutResponseHandler
```